dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunction.cs

```diff
@@ -87,21 +88,53 @@ internal KernelFunction(string name, string description, IEnumerable<AIRequestSe
         using var activity = s_activitySource.StartActivity(this.Name);
         ILogger logger = kernel.LoggerFactory.CreateLogger(this.Name);
 
-        logger.LogInformation("Function invoking.");
+        if (logger.IsEnabled(LogLevel.Trace))
```

Comment:
This is ok. I've personally only done the check when there's additional work being done at the call site, like passing arguments that'll result in a params array allocation.